### PR TITLE
Use prometheus/tsdb in ingester.

### PR DESCRIPTION
**Pull request by tomwilkie**
This PR is the first attempt at using prometheus/tsdb in the ingester.

There is lots missing/TODO:
- [ ] Add series limits to TSDB to match existing Cortex limits.
- [ ] Upstream TSDB changes for exposing chunks on series interface.
- [ ] Figure out how to do per user metrics for TSDB (number of series, ingestion rate etc).
- [ ] Ensure TSDB errors for things like out-of-order etc are being handled correctly.
- [ ] Figure out how to handle transfers for TSDB.
- [ ] Figure out how to do upgrades.
- [ ] Implement flushing TSDB to object store
- [ ] Allow index entries to point to sub-objects
- [ ] Benchmark Bigtable vs in-object-store index.

### CODE REVIEW

**Comment by codesome** [comment]
I can pick TSDB changes to be made 1 by 1, starting with series limit.

**Comment by codesome** [comment]
Limit series in TSDB: https://github.com/prometheus/tsdb/pull/617

**Comment by cstyan** [comment]
`Figure out how to do per user metrics for TSDB (number of series, ingestion rate etc).` Could we achieve this by just having a separate TSDB data dir per user ID?

**Comment by codesome** [comment]
@cstyan Yes, after I opened the PR in TSDB to have limits on number of series, it looks unnecessary to have it in TSDB. We should be able to handle that in cortex (with or without TSDB per user ID). 
Also, wouldn't it be too many database instances to handle if we have TSDB per user ID?

File: vendor/github.com/prometheus/tsdb/querier.go [hunk 1]
```diff
@@ -53,6 +53,9 @@ type Series interface {
 
 	// Iterator returns a new iterator of the data of the series.
 	Iterator() SeriesIterator
+
+	// Chunks returns a copy of the compressed chunks that make up this series.
+	Chunks() []chunks.Meta
```
Comment (bwplotka) [inline vendor/github.com/prometheus/tsdb/querier.go]: Where is this coming from? I cannot see this on tip of TSDB :thinking: nor 0.8.0 which `go mod` vendors here. 

Is this manually modifed TSDB code in vendor? Cannot see any `replace` directive on `go.mod` as well so you don't use fork as well...

Comment (tomwilkie) [reply vendor/github.com/prometheus/tsdb/querier.go]: I just added it here; opening a TSDB PR now.

Comment (tomwilkie) [reply vendor/github.com/prometheus/tsdb/querier.go]: https://github.com/prometheus/tsdb/pull/659

**Comment by tomwilkie** [comment]
Closing in favour of #1695.

**Pull request closed without merging**